app/lib/data/common_services/models/network/api_response.dart

```diff
@@ -0,0 +1,23 @@
+abstract class ApiResponse<T> {}
+
+class Success<T> extends ApiResponse<T> {
+  Success({required this.data});
+
+  T? data;
+}
+
+class Failure<T> extends ApiResponse<T> {
+  Failure({required this.failureType, this.error});
+  FailureType failureType;
+
+  /// if status code is none of 400, 401, 403, 500
+  /// we return status code only as error
+  /// otherwise, leave it null
```

Comment:
This is just a minor detail. I am very picky with documentation. It should always be:

* full sentences
* correct punctuation
* concise

Hence, it could be something like:

```suggestion
  /// Contains the http status code for unknown errors, otherwise it is null.
  String? errorMsg;
```